src/linear_algebra/matrix/hermitian.lean

```diff
@@ -139,34 +109,91 @@ lemma is_hermitian_from_blocks_iff
        congr_arg to_blocks₁₂ h, congr_arg to_blocks₂₂ h⟩,
  λ ⟨hA, hBC, hCB, hD⟩, is_hermitian.from_blocks hA hBC hD⟩
 
-end non_unital_semiring
+end has_involutive_star
 
-section semiring
+section add_monoid
+variables [add_monoid α] [star_add_monoid α] [add_monoid β] [star_add_monoid β]
 
-variables [semiring α] [star_ring α] [semiring β] [star_ring β]
+/-- A diagonal matrix `diagonal v` is hermitian if the entries have the trivial `star` operation
+(such as on the reals). -/
+@[simp] lemma is_hermitian_diagonal [has_trivial_star α] [decidable_eq n] (v : n → α) :
+  (diagonal v).is_hermitian :=
```

Comment:
Can you add the following lemma which seems like it should also be useful:
```
lemma is_hermitian_diagonal_of_star_eq [decidable_eq n] {v : n → α} (hv : star v = v) :
  (diagonal v).is_hermitian := sorry
```

Reply:
Done, but stated with `is_self_adjoint` instead.